doc/specs/Keyboard-Selection.md

```diff
@@ -0,0 +1,143 @@
+
+
+## UI/UX Design
+
+### Keybindings
+
+Thanks to Keybinding Args, there will only be 3 new commands that need to be added:
+| Action | Keybinding Args | Description |
+|--|--|--|
+| `moveSelectionAnchor` |                                                               | If a selection exists, moves the last selection anchor.
+|                       | `Enum direction { up, down, left, right}`                     | The direction the selection will be moved in. |
+|                       | `Enum expansionMode { cell, word, line, viewport, buffer }`   | The context for which to move the selection anchor to. (defaults to `cell`)
+| `selectEntireBuffer`  | | Select the entire text buffer.
+| `toggleMarkMode`      | | Enter or exit mark mode. This allows you to create an entire selection using only the keyboard. |
+
+By default, the following keybindings will be set:
+```JS
+// Cell Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "down" },  "keys": ["shift+down"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "up" },    "keys": ["shift+up"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "left" },  "keys": ["shift+left"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "right" }, "keys": ["shift+right"] },
+
+// Word Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "left",    "expansionMode": "word" }, "keys": ["ctrl+shift+left"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "right",   "expansionMode": "word" }, "keys": ["ctrl+shift+right"] },
+
+// Viewport Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "left",    "expansionMode": "viewport" }, "keys": ["shift+home"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "right",   "expansionMode": "viewport" }, "keys": ["shift+end"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "up",      "expansionMode": "viewport" }, "keys": ["ctrl+shift+up"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "down",    "expansionMode": "viewport" }, "keys": ["ctrl+shift+down"] },
+
+// Buffer Corner Selection
+{ "command": { "action": "moveSelectionAnchor", "direction": "up",      "expansionMode": "buffer" }, "keys": ["ctrl+shift+home"] },
+{ "command": { "action": "moveSelectionAnchor", "direction": "down",    "expansionMode": "buffer" }, "keys": ["ctrl+shift+end"] },
+
+// Select All
+{ "command": "selectEntireBuffer", "keys": ["ctrl+a"] },

```

Comment:
OK, so for like the third time today. If we're going to start stealing Ctrl+keys all over the keyboard like A/M for this one, 0 for the font resize, F for search.... then why did we put paste on shift? @zadjii-msft, @DHowett-MSFT, @cinnamon-msft

Reply:
Ugh. You're right. Adding the shift modifier to these for now. That one's on me.

Leaving the comment open in case people wanna have that discussion here tho.